presto-main/src/main/java/com/facebook/presto/sql/analyzer/ExpressionAnalyzer.java

```diff
@@ -803,6 +805,22 @@ protected Type visitFunctionCall(FunctionCall node, StackableAstVisitorContext<C
                 throw e;
             }
 
+            if (node.getSortItems().isPresent()) {
```

Comment:
Not really, check this query for instance:
```
explain select 1 as nationkey from customer group by nationkey order by sum(custkey order by nationkey);
```
You should explicitly check for the reference to output column from `ORDER BY` expression